mlrun/model_monitoring/db/stores/sqldb/sql_store.py

```diff
@@ -605,13 +629,56 @@ def delete_model_endpoints_resources(self) -> None:
             # Delete last analyzed records
             self._delete_last_analyzed(endpoint_id=endpoint_id)
 
-            # Delete application results records
+            # Delete application results and metrics records
             self._delete_application_result(endpoint_id=endpoint_id)
+            self._delete_application_metrics(endpoint_id=endpoint_id)
 
             # Delete model endpoint record
             self.delete_model_endpoint(endpoint_id=endpoint_id)
 
     def get_model_endpoint_metrics(
         self, endpoint_id: str, type: mm_schemas.ModelEndpointMonitoringMetricType
     ) -> list[mm_schemas.ModelEndpointMonitoringMetric]:
-        raise NotImplementedError
+        """
+        Fetch the model endpoint metrics or results (according to `type`) for the
+        requested endpoint.
+        """
+        logger.debug(
+            "Fetching metrics for model endpoint",
+            project=self.project,
+            endpoint_id=endpoint_id,
+            type=type,
+        )
+        if type == mm_schemas.ModelEndpointMonitoringMetricType.METRIC:
+            self._init_application_metrics_table()
+            table = self.application_metrics_table
+            name_col = mm_schemas.MetricData.METRIC_NAME
+        else:
+            self._init_application_results_table()
+            table = self.application_results_table
+            name_col = mm_schemas.ResultData.RESULT_NAME
+
+        # Note: the block below does not use self._get, as we need here all the
+        # results, not only `one_or_none`.
+        with sqlalchemy.orm.Session(self._engine) as session:
+            metric_rows = (
+                session.query(table)  # pyright: ignore[reportOptionalCall]
+                .filter(table.endpoint_id == endpoint_id)
+                .all()
+            )
+
+        return [
+            mm_schemas.ModelEndpointMonitoringMetric(
```

Comment:
I’m not sure if this is the right PR, but yesterday you mentioned that the result kind is being returned to the UI. I don't see it here. I noticed it’s being returned from the TSDB, but it seems more appropriate to return it from the /metrics endpoint since it’s metadata.

Reply:
"kind" isn't designed to be in this response, see section 2-c in the description of [ML-6119](https://iguazio.atlassian.net/browse/ML-6119).

Reply:
The design can be changed. Don't you think it's more accurate the way I mentioned?

Reply:
I think that's a valid point. I would also like to send the latest status in case of a result, to indicate which results should be inspected.
However, we discussed showing the status in the dropdown, and we deferred it (to >=1.8), so I think we shouldn't reopen now the API.